huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go

```diff
@@ -0,0 +1,496 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const (
+	// VaultTypeServer is the object type of the Cloud Server Backups.
+	VaultTypeServer = "server"
+	// VaultTypeDisk is the object type of the Cloud Disk Backups.
+	VaultTypeDisk = "disk"
+	// VaultTypeTurbo is the object type of the SFS Turbo Backups.
+	VaultTypeTurbo = "turbo"
+
+	// ResourceTypeServer is the type of the Cloud Server resources to be backed up.
+	ResourceTypeServer = "OS::Nova::Server"
+	// ResourceTypeDisk is the type of the Cloud Disk resources to be backed up.
+	ResourceTypeDisk = "OS::Cinder::Volume"
+	// ResourceTypeTurbo is the type of the SFS Turbo resources to be backed up.
+	ResourceTypeTurbo = "OS::Sfs::Turbo"
+)
+
+var ResourceType map[string]string = map[string]string{
+	VaultTypeServer: ResourceTypeServer,
+	VaultTypeDisk:   ResourceTypeDisk,
+	VaultTypeTurbo:  ResourceTypeTurbo,
+}
+
+func ResourceCBRVaultV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCBRVaultV3Create,
+		Read:   resourceCBRVaultV3Read,
+		Update: resourceCBRVaultV3Update,
+		Delete: resourceCBRVaultV3Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				//If the validation content has changed, please update the resource type map.
+				ValidateFunc: validation.StringInSlice([]string{
+					VaultTypeServer, VaultTypeDisk, VaultTypeTurbo,
+				}, false),
+			},
+			"consistent_level": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"crash_consistent", "app_consistent",
+				}, false),
+			},
+			"protection_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"backup", "replication",
+				}, false),
+			},
+			"size": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(1, 10485760),
+			},
+			"auto_expand": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"resources": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"server_id": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
add `ConflictsWith` field?

Reply:
Nested maps, sets and lists (greater than one length). do not support `ConflictsWith`.

Reply:
https://github.com/hashicorp/terraform-plugin-sdk/issues/71

Reply:
Ohh, I forgot it.